### PR TITLE
Feature/refacto ref man

### PR DESCRIPTION
**Pull request by jrobinAV**
Waiting for https://github.com/Avaiga/taipy-enterprise/pull/477 and https://github.com/Avaiga/taipy/pull/1851 to be merged first.

# DONE

Improvement of the Refman

- Add the Class in the navigation
- Make each entry (Classes and functions) page prettier
- Remove duplicate Attribute section
- handle page titles
- RefmanStep refactoring

image

# TODO

- Remaining bug on refmans/reference/pkg_taipy/pkg_gui/pkg_builder/Page/index.html
This is due to a postprocessing of a Xref

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -118,6 +118,7 @@ mkdocs.yml
 /docs/refmans/gui/viselements/corelements/*.md
 !/docs/userman/gui/gui_example.ipynb
 /docs/userman/xrefs
```
Comment (FabienLelaquais) [inline .gitignore]: Remove

File: Pipfile [hunk 2]
```diff
@@ -32,7 +32,7 @@ opentelemetry-sdk = ">=1.20.0"
 pandas = "==1.3.5"
 pika = "==1.3.2"
 polars = "==1.2.1"
-pyarrow = "==14.0.2"
+pyarrow = "*"
```
Comment (FabienLelaquais) [inline Pipfile:37]: Why??
This file is regenerated anyway

File: taipy_theme/assets/stylesheets/avaiga.css [hunk 3]
```diff
@@ -56,7 +56,7 @@
   /* Animations */
   --transition-duration-base: 400ms;
   --tooltip-show-delay: 400ms;
-  
+
```
Comment (FabienLelaquais) [inline taipy_theme/assets/stylesheets/avaiga.css]:
```suggestion
```

File: taipy_theme/assets/stylesheets/avaiga.css [hunk 4]
```diff
@@ -429,7 +428,7 @@
 }
 
 /* Display the right icon for the right theme */
-.icon-light{ 
+.icon-light{
```
Comment (FabienLelaquais) [inline taipy_theme/assets/stylesheets/avaiga.css]:
```suggestion
.icon-light {
```

File: taipy_theme/assets/stylesheets/avaiga.css [hunk 5]
```diff
@@ -1019,7 +1018,7 @@ div.tp-nav-button-group {
     display: flex;
     gap: 0.1rem;
     margin: 5px 0 5px 0;
-    width: calc(100% - .1rem);
+    width: calc(100% - 3rem);
```
Comment (FabienLelaquais) [inline taipy_theme/assets/stylesheets/avaiga.css:1020]: This must match the spacing between the two horizontal buttons in the nav section.
Are you really sure??

File: tools/postprocess.py [hunk 6]
```diff
@@ -125,7 +125,7 @@ def create_button(
         elif group == "end-group":
                 return html + "</div>"
         else:
-            return html
+            return "<div class=\"tp-nav-button-single\">" + html + "</div>"
```
Comment (FabienLelaquais) [inline tools/postprocess.py:128]: This I think (I mean the whole layout) should be fixed in raw CSS as before.
May be I'll change that.

File: tools/postprocess.py [hunk 7]
```diff
@@ -369,6 +372,8 @@ def on_post_build(env):
                         + r"(\">.*?</a>)"
                     )
                     for xref in FREE_XREF_RE.finditer(html_content):
+                        print("------------ THIS IS NEVER PRINTED !!!! ------------------")
```
Comment (FabienLelaquais) [inline tools/postprocess.py]: So drop these two lines please.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Minor but still

File: taipy_theme/assets/stylesheets/avaiga.css (hunk 5 above)
Comment (jrobinAV) [reply taipy_theme/assets/stylesheets/avaiga.css:1020]: I am not

File: tools/postprocess.py (hunk 6 above)
Comment (jrobinAV) [reply tools/postprocess.py:128]: I am not at all comfortable with any of the CSS, HTML, or postprocessing. So I fully trust you. 
I propose to merge it as it is and let you handle this change.

**Pull request merged**